### PR TITLE
Fixes for raw arrays

### PR DESCRIPTION
**Pull request by marchdf**
This is to fix raw array/pointers and memory leaks.

### CODE REVIEW

**Review by jrood-nrel — APPROVED** [review]

**Comment by marchdf** [comment]
Still need to test by compiling

**Review by mbkuhn — APPROVED** [review]
well done!

File: src/modes_hosgrid.cpp [hunk 1]
```diff
@@ -98,12 +96,12 @@ void modes_hosgrid::populate_hos_eta_nondim(
     const int n0, const int n1, fftw_plan p, fftw_complex *eta_modes,
     amrex::Gpu::DeviceVector<amrex::Real> &HOS_eta) {
   // Local array for output data
-  double out[n0 * n1];
+  amrex::Vector<double> out(n0 * n1, 0.0);
```
Comment (jrood-nrel) [inline src/modes_hosgrid.cpp]: I'm seeing `double` all around here. Are they required?

Comment (marchdf) [reply src/modes_hosgrid.cpp]: as opposed to "amrex::Real"? Yeah idk what FFT expects

Comment (marchdf) [reply src/modes_hosgrid.cpp]: didn't really want to deal with that I guess?

Comment (jrood-nrel) [reply src/modes_hosgrid.cpp]: oh yeah FFTW, let's just fix it for now

Comment (jrood-nrel) [reply src/modes_hosgrid.cpp]: I mean not convert to amrex::Real

**Pull request merged**